javascripts/2048.js

```diff
@@ -1,32 +1,276 @@
 $(document).ready(function() {
   console.log('ready!');
+  arrayInitialize();
+  initializeGame();
   $('body').keydown(function(event){
     var arrow_keys = [37, 38, 39, 40];
+
     if(arrow_keys.indexOf(event.which) > -1) {
-      var tile = $('.tile');
-      moveTile(tile, event.which);
+      // var tile = $('.tile');
+      moveTile(event.which);
       event.preventDefault();
     }
   })
 })
 
-function moveTile(tile, direction) {
+function pop(tile) {
+  $(tile).addClass("popper")
+  .on("animationend",function() {
+    $(this).removeClass("popper");
+  });
+}
+
+
+ //keeps track of where on the board has a tile
+var tile_array = [];
+//set up for 2D array
+function arrayInitialize(){
+  for(var i = 0; i < 4; i++) {
```

Comment:
Javascript allows you to create an empty allocated array, and also allows you to fill arrays with predetermined values. So a cute way of combining `arrayInitialize` and `setUndefinedTile` is:
```
function arrayInitialize(){
  tile_array = new Array(16);
  tile_array.fill([void 0, void 0]);
}
```
This will produce an array with 16 slots, then fill each slot with an array of [undefined, undefined]